flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1337,6 +1350,29 @@ def test_resize_destroyed_volume(self):
         )
         self.assertEqual(exception.args, (volume.blockdevice_id,))
 
+    def test_foreign_cluster_volume(self):
+        """
+        Test that list_volumes() excludes volumes belonging to
+        other Flocker clusters.
+        """
+
+        flocker_volume1 = self.api.create_volume(
+            dataset_id=uuid4(),
+            size=REALISTIC_BLOCKDEVICE_SIZE,
+        )
+
+        old_cluster_id = self.api.cluster_id
```

Comment:
Thanks for pointing out! Reworked this part to SkipTest if cluster_id does not exist. @exarkun , is there a plan for adding cluster_id for LoopbackBlockDeviceAPI? If yes, i can remove SkipTest for test_foreign_cluster_volume when cluster_id is available. Thanks!